### ISSUE TITLE
Does not work at all (Linux)

### ISSUE THREAD

**Issue by UnusedEngineWasTaken** [p3tr1ch0r/insomniaHUD#6]
I have added the folder into my custom folder as per usual, and also had changed the name to insomniaHUD and yet it still opens with the default HUD rather than the custom HUD

**Comment by p3tr1ch0r** [comment]
very very strange. i'll fire up a virtual machine later to test it out. care to tell me what distribution you are using, alongside your display manager and desktop environment and/or window manager?

**Comment by UnusedEngineWasTaken** [comment]
im using arch linux, lightdm and i3

**Comment by UnusedEngineWasTaken** [comment]
please also do keep in mind i have downgraded my `libx11` and `lib32-libx11` to versions 1.8.2-2 as to stop TF2 randomly crashing (only on Nvidia gpus)

**Comment by p3tr1ch0r** [comment]
Unfortunately, I am unable to reproduce this. I'll try and get some other Linux users to test the HUD out, but as of right now, I do not see this getting fixed anytime soon. My apologies.

**Comment by The-Legendary-Bean** [comment]
I am planning to distro hop to void linux and will update you if it works there

**Comment by KaIeetan** [comment]
TF2 huds will not work on Linux unless the folder name is completely lowercase (i.e. "insomniahud" instead of the required "insomniaHUD")

**Comment by The-Legendary-Bean** [comment]
> TF2 huds will not work on Linux unless the folder name is completely lowercase (i.e. "insomniahud" instead of the required "insomniaHUD")

Ill try it and see
Thank you for your time :)

**Comment by Viceroyy** [comment]
> TF2 huds will not work on Linux unless the folder name is completely lowercase (i.e. "insomniahud" instead of the required "insomniaHUD")

tf2 on linux is very picky about mods, it really dislikes uppercase letters and spaces, just get rid of them. e.g. "Some Mod" -> "some_mod".

**Comment by The-Legendary-Bean** [comment]
Its working now, thank you!!
I also wanted to tell you that this HUD is amazing!!!
I look forward to the future of this HUD!

**Comment by The-Legendary-Bean** [comment]
oh wait this js my alt

**Comment by PoorPocketsMcNewHold** [comment]
> [**@Viceroyy**](https://github.com/Viceroyy) commented on [Feb 26, 2023, 1:45 PM UTC](https://github.com/p3tr1ch0r/insomniaHUD/issues/6#issuecomment-1445365763):
> 
> > > TF2 huds will not work on Linux unless the folder name is completely lowercase (i.e. "insomniahud" instead of the required "insomniaHUD")
> > 
> > tf2 on linux is very picky about mods, it really dislikes uppercase letters and spaces, just get rid of them. e.g. "Some Mod" -> "some\_mod".

It’s not that it’s picky, it’s just because most Linux filesystems, are case-sensitive compared to Windows ones. Where a folder named SOMETHING and one named something won’t make a difference at all on Windows, it will on Linux. The same thing applies to files, including .vpk ones.

Team Fortress 2 does recognize uppercase mods, but doesn’t correct them properly, mainly recreating a lowercase empty folder of it automatically, without actually moving/copying the files. It could also be due to some case, where the game list internally modded files to read from as lowercase, once it finished crawling throught the entire folder, and as such, recreate incorrect lowercases ones instead of the read Uppercase one.

Tl;dr : Unix-type file systems are case-sensitive, Windows isn’t, the game probably isn’t either, and it may just default everything to lowercase ″because this behavior doesn’t matter on Windows″.

@UnusedEngineWasTakencorrected? Care to close this issue if it has been corrected?
